src/modules/Staff/Service.php

```diff
@@ -695,4 +695,25 @@ public function authorizeAdmin($email, $plainTextPassword)
 
         return $this->di['auth']->authorizeUser($model, $plainTextPassword);
     }
+
+    public function sendPasswordResetConfirmation($hash)
+    {
+        $reset = $this->di['db']->findOne('ClientPasswordReset', 'hash = ?', [$hash]);
+        if (!$reset instanceof \Model_ClientPasswordReset) {
+            throw new \Box_Exception('The link have expired or you have already confirmed password reset.');
+        }
+        $new_pass = $this->di['tools']->generatePassword();
+        $c = $this->di['db']->getExistingModelById('Admin', $reset->client_id, 'Client not found');
+        $c->pass = $this->di['password']->hashIt($new_pass);
+        $this->di['db']->store($c);
+
+        // send email
+        $email = [];
+        $email['to_client'] = $reset->client_id;
+        $email['code'] = 'mod_staff_password_reset_approve';
+        $email['password'] = $new_pass;
```

Comment:
If possible we should allow the user make / create the password instead sending. 

(Same should also apply to the normal user off course)

Reply:
Should I be attempting this one time?

Reply:
If you'd like, but it doesn't need to be done as part of this PR and it's fine if it's done at a later date

Reply:
Ok cool. I'll finish the admin reset before I do anything else.